pennylane/devices/qubit/apply_operation.py

```diff
@@ -198,6 +199,31 @@ def _(op: type_op, state):
     return _apply_operation_default(op, state, is_state_batched, debugger)
 
 
+@apply_operation.register
+def apply_mid_measure(op: MidMeasureMP, state, is_state_batched: bool = False, debugger=None):
+    """Applies a native mid-circuit measurement."""
+    from .measure import measure  # pylint: disable=import-outside-toplevel
+
+    if is_state_batched:
+        raise ValueError("MidMeasureMP cannot be applied to batched states.")
+    wire = op.wires
+    probs = measure(qml.probs(wire), state)
+    dark_branch = np.random.binomial(1, probs[0])
+    axis = wire.toarray()[0]
+    slices = [slice(None)] * state.ndim
+    slices[axis] = dark_branch
+    state[tuple(slices)] = 0.0
+    state_norm = np.linalg.norm(state.ravel())
```

Comment:
No need to flatten the state. numpy uses the frobenius norm by default, which is effectively the L2 vector norm of the flattened array

```suggestion
    state_norm = np.linalg.norm(state)
```